benchmarks/expected-plans/q19.txt

```diff
@@ -1,9 +1,8 @@
 Projection: SUM(lineitem.l_extendedprice * Int64(1) - lineitem.l_discount) AS revenue
   Aggregate: groupBy=[[]], aggr=[[SUM(CAST(lineitem.l_extendedprice AS Decimal128(38, 4)) * CAST(Decimal128(Some(100),23,2) - CAST(lineitem.l_discount AS Decimal128(23, 2)) AS Decimal128(38, 4))) AS SUM(lineitem.l_extendedprice * Int64(1) - lineitem.l_discount)]]
-    Projection: lineitem.l_extendedprice, lineitem.l_discount
```

Comment:
This is not really redundant I think (when converting to a physical aggregation, we don't want to hash repartition the other columns)?

Reply:
Hm, I don't think the hash aggregation takes columns into account which are neither in the `groupBy` nor in the `aggr` expressions. Even if it did, it probably shouldn't be the job of `CommonSubexprEliminate` to ensure that redundant columns are projected out first, that should be it's own (physical?) optimization rule IMO.